modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -967,808 +967,6 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
-
-                for( int cn0 = 0; cn0 < inpCn; cn0 += blk_size_cn )
-                {
-                    int cn1 = std::min(cn0 + blk_size_cn, inpCn);
-                    int ncn = cn1 - cn0, vsz = karea*ncn;
-                    int vsz_a = (int)alignSize(vsz, valign);
-                    const float* wptr = wptr_orig + cn0*karea;
-                    // we apply [Channels][P]ReLU (if any) during the final pass only.
-                    const float* relu = cn1 == inpCn && reluptr_ ? reluptr_ + startOutCn : 0;
-
-                    for( int ofs0 = stripeStart; ofs0 < stripeEnd; ofs0 += blk_size )
-                    {
-                        int ofs, ofs1 = std::min(ofs0 + blk_size, stripeEnd);
-                        int bsz = ofs1 - ofs0;
-
-                        int out_d = ofs0 / (outH * outW);
-                        int out_i = (ofs0 - out_d * outH * outW) / outW;
-                        int out_j = ofs0 % outW;
-
-                        if (depthWiseConvolution)
-                        {
-                            CV_Assert(out_i == 0 && out_j == 0);
-                            int in_d = out_d * stride_d - pad_d;
-                            const float* inptr_ = data_inp0 + (cn0*depth*height + in_d*height)*width;
-                            float* outptr_ = data_out0 + ofs0;
-
-                        #if CV_TRY_AVX2
-                            if(useAVX2)
-                                opt_AVX2::fastDepthwiseConv(wptr, kernel_h, kernel_w,
-                                    stride_h, stride_w, dilation_h, dilation_w, pad_t, pad_l,
-                                    biasptr, relu, inptr_, height, width, outptr_, out_d, outH, outW);
-                            else
-                        #endif
-                        #if CV_TRY_AVX
-                            if(useAVX)
-                                opt_AVX::fastDepthwiseConv(wptr, kernel_h, kernel_w,
-                                    stride_h, stride_w, dilation_h, dilation_w, pad_t, pad_l,
-                                    biasptr, relu, inptr_, height, width, outptr_, out_d, outH, outW);
-                            else
-                        #endif
-                        #if CV_TRY_RVV
-                            if(useRVV)
-                                opt_RVV::fastDepthwiseConv(wptr, kernel_h, kernel_w,
-                                    stride_h, stride_w, dilation_h, dilation_w, pad_t, pad_l,
-                                    biasptr, relu, inptr_, height, width, outptr_, out_d, outH, outW);
-                            else
-                        #endif
-                        #if CV_TRY_LASX
-                            if(useLASX)
-                                opt_LASX::fastDepthwiseConv(wptr, kernel_h, kernel_w,
-                                    stride_h, stride_w, dilation_h, dilation_w, pad_t, pad_l,
-                                    biasptr, relu, inptr_, height, width, outptr_, out_d, outH, outW);
```

Comment:
We need performance comparison in PR's description for that massive changes.
Also it makes sense to compare memory consumption too (to avoid 3x regressions like recent case).

Reply:
Hi @alalek. Thanks for your reminder, the purpose of this PR is that let Conv3d and Conv1D execute in the new branch. And it will not affect the speed and memory consumption of Conv2D. (Conv2D Related PR: https://github.com/opencv/opencv/pull/21910. )

I will update the speed performance test and the memory consumption of Conv3D and Conv1D.
In theory, Conv3D and Conv1D require twice as much memory as before, since we need repack the weight at the `fast_conv` initialized stage. It will not reach 7 times (because Conv3D and Conv1D do not support Winograd).

Reply:
Please use OpenCV performance testing infrastructure instead of some kind of manual testing:
- Single result is not enough. We must ensure that there are no regression in other cases.
- `Resnet34_kinetics` - no such test in opencv_perf_dnn
- CPU/SIMD code optimization validation should be done with `--perf_threads=1`
- i7-12700k has P and E cores. Need to bind test process to selected CPUs.

---

Finally, we have regressions, in conv1d case:

|Name of Test|base|patch|x-factor|
|---|:-:|:-:|:-:|
|conv1d::Conv1D::(GFLOPS=0.000, K=[3], IN={1, 2, 19}, OCN=2, G=2, S=2, P=(1, 1), BIAS, OCV/CPU)|0.001|0.001|0.74|
|conv1d::Conv1D::(GFLOPS=0.000, K=[3], IN={1, 2, 25}, OCN=2, G=2, P=(2, 2), PM=SAME, OCV/CPU)|0.001|0.001|0.72|
|conv1d::Conv1D::(GFLOPS=0.000, K=[3], IN={1, 6, 10}, OCN=6, PM=VALID, BIAS, OCV/CPU)|0.001|0.001|0.74|

(CPU: i7-12700k, Linux)

Test launch cmd:

```
taskset -c 0 ./bin/opencv_perf_dnn '--gtest_filter=*' --gtest_output=xml:../perf/pr22905/0-1th.xml --perf_threads=1
```

Report cmd:

```
python3 ${OPENCV_SRC}/modules/ts/misc/summary.py -m min -f conv1d ../perf/pr22905/{0,1}-1th.xml
```

(use `-o markdown` for GitHub report)

Reply:
> i7-12700k has P and E cores. Need to bind test process to selected CPUs.

Thanks for your reminder. How about I add new performance with a single thread result?

> conv1d::Conv1D

This case is too small, I think it is not a good example to reflect the speedup.

Reply:
Performance report is not intended to show non-reproducible marketing **single** number.
Performance report is required to track regressions and to avoid them. We need all numbers from performance tests.

@vpisarev Used pipeline of optimization development has serious gaps. Ignoring existence of performance tests during optimization is not an acceptable flow. We need to fix that process.

Reply:
I will try to update the performance test resulting from performance testing infrastructure.

Because the calculation amount of Conv1D in the performance test is too small, I am worried that the test results cannot reflect the model in the real application scenario.